internal/services/compute/snapshot_resource.go

```diff
@@ -91,6 +92,15 @@ func resourceSnapshot() *pluginsdk.Resource {
 				Default:      string(snapshots.NetworkAccessPolicyAllowAll),
 			},
 
+			"disk_access_id": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
```

Comment:
Two things:

1. Since the test isn't checking the update codepath, can this be updated?
2. Can we also add validation for this field to ensure the correct value is provided:

```suggestion
				Optional: true,
				ValidateFunc: diskaccesses.ValidateDiskAccessID,
```